src/Microsoft.SqlTools.ServiceLayer/TableDesigner/TableDesignerValidator.cs

```diff
@@ -555,4 +556,22 @@ public List<TableDesignerIssue> Run(Dac.TableDesigner designer)
             return errors;
         }
     }
+
+    public class MemoryOptimizedCannotBeEnabledWhenNotSupportedRule : ITableDesignerValidationRule
+    {
+        public List<TableDesignerIssue> Run(Dac.TableDesigner designer)
+        {
+            var table = designer.TableViewModel;
+            var errors = new List<TableDesignerIssue>();
+            if (!designer.IsMemoryOptimizedTableSupported && designer.TableViewModel.IsMemoryOptimized)
+            {
+                errors.Add(new TableDesignerIssue()
+                {
+                    Description = string.Format("Memory-optimized is either not supported or not enabled in this database."),
```

Comment:
Not sure if I've asked this before, but do we need to use localized string for all these validation descriptions?

Reply:
I was planning to move all the rules to dacfx, so I left these unlocalized. I'll create an issue in Aug milestone to track this task.